pkg/agent/client.go

```diff
@@ -294,6 +295,7 @@ func (a *AgentClient) Serve() {
 				}
 				continue
 			}
+			metrics.Metrics.ObserveDialLatency(time.Since(start))
```

Comment:
So this looks like it won't create a metric for an error, even a timeout. Is that desired?

Reply:
That's intentional. I want to avoid dial timeout from bloating the average latency. Also this is consistent with the egressselector [metrics](https://github.com/kubernetes/kubernetes/blob/6079cebfae672335e958e03e7b0c258ba264d28f/staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go#L225-L239) in k/k.

Also note that the metrics in k/k egressslector package covers the end-to-end dial error.